Sources/LiftNavigationController.swift

```diff
@@ -0,0 +1,100 @@
+        self.topViewController = topViewController
+
+        super.init(nibName: nil, bundle: nil)
+
+        self.bottomScrollView.bottomViewControllers = bottomViewControllers
+
+        self.topViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        self.view.translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.view.addGestureRecognizer(self.gestureRecognizer)
+        self.addSubviewsAndConstraints()
+    }
+
+    func addSubviewsAndConstraints() {
+        self.addChildViewController(self.topViewController)
+
+        self.view.addSubview(self.topViewController.view)
+        self.view.addSubview(self.navigationBar)
+        self.view.addSubview(self.bottomScrollView)
+
+        self.heightAnchor = self.topViewController.view.topAnchor.constraint(equalTo: self.view.topAnchor, constant: 0.0)
+        self.heightAnchor?.isActive = true
+
+        self.topViewController.view.leftAnchor.constraint(equalTo: self.view.leftAnchor).isActive = true
+        self.topViewController.view.widthAnchor.constraint(equalTo: self.view.widthAnchor).isActive = true
+        self.topViewController.view.heightAnchor.constraint(equalToConstant: self.view.bounds.height - LiftNavigationController.navigationBarHeight).isActive = true
+
+        self.navigationBar.topAnchor.constraint(equalTo: self.topViewController.view.bottomAnchor).isActive = true
+        self.navigationBar.leftAnchor.constraint(equalTo: self.view.leftAnchor).isActive = true
+        self.navigationBar.widthAnchor.constraint(equalTo: self.view.widthAnchor).isActive = true
+        self.navigationBar.heightAnchor.constraint(equalToConstant: LiftNavigationController.navigationBarHeight).isActive = true
+
+        self.bottomScrollView.topAnchor.constraint(equalTo: self.navigationBar.bottomAnchor).isActive = true
+        self.bottomScrollView.leftAnchor.constraint(equalTo: self.view.leftAnchor).isActive = true
+        self.bottomScrollView.widthAnchor.constraint(greaterThanOrEqualTo: self.view.widthAnchor).isActive = true
+        self.bottomScrollView.heightAnchor.constraint(equalTo: self.view.heightAnchor, constant: -LiftNavigationController.navigationBarHeight).isActive = true
+    }
+
+    func didSwipe(_ recognizer: UISwipeGestureRecognizer) {
+        if recognizer.direction == .up {
+            self.heightAnchor?.constant = -(self.view.bounds.height - LiftNavigationController.navigationBarHeight)
+            self.gestureRecognizer.direction = .down
+        } else {
+            if recognizer.direction == .down {
+                self.heightAnchor?.constant = 0.0
+                self.gestureRecognizer.direction = .up
+            }
+        }
+        UIView.animate(withDuration: 0.2, delay: 0.0, options: [.curveEaseOut], animations: {
```

Comment:
I usually add `.allowUserInteraction, .beginFromCurrentState` to animations, makes them smoother.